cmd/copy.go

```diff
@@ -184,10 +183,48 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 
 	cooked.fromTo = fromTo
 
-	// copy&transform flags to type-safety
-	cooked.stripTopDir = raw.stripTopDir
+	// Check if source has a trailing wildcard on a URL
+	if fromTo.From().IsRemote() {
+		// Because local already handles wildcards via a list traverser, we should only handle the trailing wildcard --strip-top-dir inference remotely.
+		// To avoid getting trapped by parsing a URL and losing a sense of which *s are real, strip the SAS token in a """unsafe""" way.
+		splitURL := strings.Split(cooked.source, "?")
+
+		// replace %2A with %00 (NULL).
```

Comment:
>//  [](start = 2, length = 3)

this part came off strange at first read, please consider adding a comment to clarify that the %2As are being replaced so that they don't get turned into * when we look at the parsed url

Reply:
Good point-- it says what it does but not why it does it.